spec/unit/lock_spec.rb

```diff
@@ -46,22 +46,22 @@ def self.debug(_); end
           with(:body => "{\"version\":1,\"concurrency\":2,\"holders\":{}}").
           to_return( status: 200, body: "true")
 
-        expect(Semaphore.get_or_create('/check-lock/my_lock', 2)).to be_a(Semaphore)
```

Comment:
why this change?

Reply:
this is a change due to version of "something" (see https://github.com/criteo-cookbooks/choregraphie/pull/41/commits/cb1a58603daa4c3217785184b3d7d0e561ee8c03)

Reply:
otherwise the url send to consul are not identical and contains `//` at the beginning of the path

Reply:
This is to me a breaking change.
Can't you strip the leading '/' in `Semaphore.get_or_create` if provided ?

Reply:
that's a change in diplomat not on my code.

Reply:
If this object is used outside your lib, that's your issue (if it is internal only, that's different), either lock diplomat or hack around it ;)

Reply:
you're right I could mock diplomat instead of mocking http calls.
I think this is outside of the scope of this review